ush/generate_FV3LAM_wflow.py

```diff
@@ -137,9 +139,23 @@ def generate_FV3LAM_wflow(
         verbose=debug,
     )
 
-    create_symlink_to_file(
-        wflow_launch_script_fp, os.path.join(exptdir, wflow_launch_script_fn), False
-    )
+    with open(wflow_launch_script_fp, "r", encoding='utf-8') as launch_script_file:
+        launch_script_content = launch_script_file.read()
+
+    # Stage an experiment-specific launch file in the experiment directory
+    template = Template(launch_script_content)
+
+    # The script needs several variables from the workflow and user sections
+    template_variables = {**expt_config["user"], **expt_config["workflow"],
+            "valid_vals_BOOLEAN": list_to_str(expt_config["constants"]["valid_vals_BOOLEAN"])}
+    launch_content =  template.safe_substitute(template_variables)
+
+    launch_fp = os.path.join(exptdir, wflow_launch_script_fn)
+    with open(launch_fp, "w", encoding='utf-8') as expt_launch_fn:
+        expt_launch_fn.write(launch_content)
+
+    os.chmod(launch_fp, os.stat(launch_fp).st_mode|stat.S_IXUSR)
```

Comment:
I'm unclear on exactly what's happening here, but this line is giving me an error specifically when I attempt to run multiple verification WE2E tests on Hera:

## Works fine:
```
./run_WE2E_tests.py -m=hera -a fv3lam -q -t MET_verification_only_vx
...
...

./run_WE2E_tests.py -m=hera -a fv3lam -q -t MET_ensemble_verification_only_vx
...
...

./run_WE2E_tests.py -m=hera -a fv3lam -q -t fundamental
...
...
```

## Fails:

```
./run_WE2E_tests.py -m=hera -a fv3lam -q -t verification
...
*********************************************************************
FATAL ERROR:
Experiment generation failed. See the error message(s) printed below.
For more detailed information, check the log file from the workflow
generation script: log.run_WE2E_tests
*********************************************************************

Traceback (most recent call last):
  File "/scratch1/BMC/hmtb/kavulich/UFS/workdir/PR_1098/ufs-srweather-app/tests/WE2E/./run_WE2E_tests.py", line 550, in <module>
    run_we2e_tests(homedir,args)
  File "/scratch1/BMC/hmtb/kavulich/UFS/workdir/PR_1098/ufs-srweather-app/tests/WE2E/./run_WE2E_tests.py", line 232, in run_we2e_tests
    expt_dir = generate_FV3LAM_wflow(ushdir,logfile=f"{ushdir}/log.generate_FV3LAM_wflow",
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/scratch1/BMC/hmtb/kavulich/UFS/workdir/PR_1098/ufs-srweather-app/tests/WE2E/../../ush/generate_FV3LAM_wflow.py", line 157, in generate_FV3LAM_wflow
    os.chmod(launch_fp, os.stat(launch_fp).st_mode|stat.S_IXUSR)
                                                   ^^^^^^^^^^^^
AttributeError: 'str' object has no attribute 'S_IXUSR'



./run_WE2E_tests.py -m=hera -a fv3lam -q -t MET_ensemble_verification_only_vx_time_lag MET_ensemble_verification_only_vx
...
*********************************************************************
FATAL ERROR:
Experiment generation failed. See the error message(s) printed below.
For more detailed information, check the log file from the workflow
generation script: log.run_WE2E_tests
*********************************************************************

Traceback (most recent call last):
  File "/scratch1/BMC/hmtb/kavulich/UFS/workdir/PR_1098/ufs-srweather-app/tests/WE2E/./run_WE2E_tests.py", line 550, in <module>
    run_we2e_tests(homedir,args)
  File "/scratch1/BMC/hmtb/kavulich/UFS/workdir/PR_1098/ufs-srweather-app/tests/WE2E/./run_WE2E_tests.py", line 232, in run_we2e_tests
    expt_dir = generate_FV3LAM_wflow(ushdir,logfile=f"{ushdir}/log.generate_FV3LAM_wflow",
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/scratch1/BMC/hmtb/kavulich/UFS/workdir/PR_1098/ufs-srweather-app/tests/WE2E/../../ush/generate_FV3LAM_wflow.py", line 157, in generate_FV3LAM_wflow
    os.chmod(launch_fp, os.stat(launch_fp).st_mode|stat.S_IXUSR)
                                                   ^^^^^^^^^^^^
AttributeError: 'str' object has no attribute 'S_IXUSR'
```
I honestly can't make heads or tails of the failure condition. It is consistent (the same sets of tests will always fail), but there doesn't seem to be any rhyme or reason to *which* sets of tests fail. This might be because I don't fully understand the behavior of this command; hopefully the fix will be obvious to you.

Reply:
After a bit of offline debugging together, we isolated the problem to the way that this script makes all flattened YAML variables available as local variables. For the verification tasks, the Rocoto YAML for the ensemble verification task defines `stat: MEAN PROB` causing a namespace collision with the imported `stat` that should have `S_IXUSR` as an attribute.

To fix this problem, I have modified the import to read:

```
from stat import S_IXUSR
```

in hopes that `S_IXUSR` won't cause a namespace collision!

Reply:
Confirmed I no longer see this error 👍